circuit/src/dynamic_sets/native.rs

```diff
@@ -974,10 +968,121 @@ mod test {
 			let den = compose_big_decimal::<Fr, NUM_LIMBS, POWER_OF_TEN>(tw.den_decomposed);
 			let ratio = BigRational::new(num.to_bigint().unwrap(), den.to_bigint().unwrap());
 			println!(
-				"real score: {:?}, is bigger than 435: {:?}",
+				"real score: {:?}, is bigger than {}: {:?}",
 				ratio.to_integer().to_str_radix(10),
+				THRESHOLD,
 				tw.is_bigger,
 			);
 		}
 	}
+
+	fn test_scaling_multi_peers<
+		const NUM_NEIGHBOURS: usize,
+		const NUM_ITERATIONS: usize,
+		const INITIAL_SCORE: u128,
+		const NUM_LIMBS: usize,
+		const POWER_OF_TEN: usize,
+		const THRESHOLD: u128,
+	>() -> bool {
+		let rng = &mut thread_rng();
+		let mut ops_raw = [(); NUM_NEIGHBOURS].map(|_| [(); NUM_NEIGHBOURS].map(|_| 0_u128));
+		for i in 0..NUM_NEIGHBOURS {
+			for j in 0..NUM_NEIGHBOURS {
+				ops_raw[i][j] = rng.gen::<u8>() as u128;
+			}
+			let idx = rng.gen::<usize>() % NUM_NEIGHBOURS;
+			ops_raw[i][idx] = 1;
+			ops_raw[idx][i] = 1;
+			ops_raw[i][i] = 0;
```

Comment:
Why are we assigning 1s and 0s at these indices?

Reply:
These are for adding more randomization.
When I try the `ops_raw[i][j] = rng.gen::<u8>() as u128;` in the opinion score generation, it often outputs 2-digit numbers in most case.